examples/api/test/widgets/scroll_view/custom_scroll_view.1_test.dart

```diff
@@ -0,0 +1,64 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/widgets/scroll_view/custom_scroll_view.1.dart' as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('What should be visible or not visible in the initial state.', (WidgetTester tester) async {
+    await tester.pumpWidget(const example.CustomScrollViewExampleApp());
+
+    expect(find.byType(Scaffold), findsOne);
+    expect(find.descendant(
+      of: find.byType(Scaffold),
+      matching: find.byType(AppBar),
+    ), findsOne);
+    expect(find.descendant(
+      of: find.byType(AppBar),
+      matching: find.byType(IconButton),
+    ), findsOne);
+    expect(find.descendant(
+      of: find.byType(Scaffold),
+      matching: find.byType(CustomScrollView),
+    ), findsOne);
+    expect(find.descendant(
+      of: find.byType(CustomScrollView),
+      matching: find.byType(SliverList),
+    ), findsOne);
```

Comment:
Thoses checks can be removed because there are checking the structure of the implementation. The goal for thoses tests is to check if the example works from a user point of view.

This can be replace by only one expect that checks if the IconButton with the plus sign is visible.
Eventually we can keep `expect(find.byType(SliverList), findsOne);` (this one might be interesting from a developper point of view, as it emphasizes the solution illustrated by this example).